### PR TITLE
Pass over Simplify CRs, 2021-10-21

### PR DESCRIPTION
**Pull request by mshinwell**
A lot of small changes.  If reviewers select "files owned by me" then the diffs should be easy enough to read.  We're now down to 36 CRs in the simplifier across 14 files.

### CODE REVIEW

**Review by Gbury — APPROVED** [review]
I've looked at all the files owned by me, and this looks good.

**Review by lthls — APPROVED** [review]
Ok for the files owned by me

**Review by poechsel — APPROVED** [review]

File: middle_end/flambda2/bound_identifiers/bound_for_function.ml [hunk 1]
```diff
@@ -42,12 +42,17 @@ let[@ocamlformat "disable"] print ppf
 
 let create ~return_continuation ~exn_continuation ~params ~my_closure ~my_depth
     =
-  let params_set = BP.Set.of_list params in
-  if List.length params <> BP.Set.cardinal params_set
+  BP.List.check_no_duplicates params;
```
Comment (Gbury) [inline middle_end/flambda2/bound_identifiers/bound_for_function.ml:45]: Shouldn't the no duplicate check be inside the `check_invariants ()` block ?
EDIT: so the `check_no_duplicates` function already polls the `check_invariant` function, but it's a bit confusing...

**Review by Gbury — APPROVED** [review]

File: middle_end/flambda2/bound_identifiers/bound_for_function.ml (hunk 1 above)
Comment (mshinwell) [reply middle_end/flambda2/bound_identifiers/bound_for_function.ml:45]: Yeah, but I think it's probably ok.

File: middle_end/flambda2/simplify/simplify_static_const.ml [hunk 2]
```diff
@@ -31,8 +31,9 @@ let simplify_field_of_block dacc (field : Field_of_static_block.t) =
     let simple = T.get_alias_exn ty in
     Simple.pattern_match simple
       ~name:(fun name ~coercion ->
-        (* CR lmaurer: This will break if you try and put a coerced thing in a
-           block *)
+        (* CR lmaurer for lmaurer: This will break if you try and put a coerced
+           thing in a block mshinwell: is that guaranteed not to occur? It isn't
+           obvious to me why this would be the case. *)
```
[the remark this answers is not in the document]
Comment (mshinwell) [reply middle_end/flambda2/simplify/simplify_static_const.ml:36]: For the moment I'm going to leave this; if the assertion fails, we can revisit.

**Pull request merged**